frontend/src/components/pages/admin/AssessmentEditorPage.tsx

```diff
@@ -60,25 +68,34 @@ const AssessmentEditorPage = (): React.ReactElement => {
     variables: { id: state?.id },
   });
 
+  const isLoading = loadingCreate || loadingUpdate || loadingDelete;
+
+  const methods = useForm<TestRequest>();
   const {
-    handleSubmit,
     register,
-    formState: { errors },
+    formState: { errors, isDirty: isFormDirty },
     control,
     setValue,
     watch,
     clearErrors,
-  } = useForm<TestRequest, unknown>({
-    defaultValues: {
-      name: state?.name,
-      questions: state?.questions,
-      grade: state?.grade,
-      assessmentType: state?.assessmentType,
-      status: state?.status,
-      curriculumCountry: state?.curriculumCountry,
-      curriculumRegion: state?.curriculumRegion,
-    },
-  });
+    reset: resetForm,
+  } = methods;
+  useEffect(() => {
+    if (state) {
+      resetForm({
+        name: state.name,
+        questions: state.questions,
+        grade: state.grade,
+        assessmentType: state.assessmentType,
+        status: state.status,
+        curriculumCountry: state.curriculumCountry,
+        curriculumRegion: state.curriculumRegion,
+      });
+      setQuestions(state.questions);
+    }
+  }, [state, resetForm]);
+  const isDirty =
+    !redirectTo && (isFormDirty || questions !== state?.questions);
```

Comment:
btw i think this doesn't work for a new test bc `questions = []` but `state?.questions = undefined`. could we handle that case?

Reply:
@carissa-tang I thought that for new tests we said we would always display the reload prompt?

Reply:
oh ok sure